### PR TITLE
Branch coverage

### LINKED ISSUE #32: 100% Branch Coverage
Somewhere in applyDeposits, our unit test do not cover all possible branches. I suspect that it has to do with the requirement that slots be applied sequentially (since this condition has an OR and we may not have tested both scenarios).

### PR DESCRIPTION
**Pull request by bh2smith**
- Finally achieved Full Coverage on Unit Tests
- Also using truffle-assertions to catch revert messages

Closes #32 and #39

### CODE REVIEW

**Comment by bh2smith** [comment]
Note that this is based on the claim withdraw script branch (which should be merged first).

File: package.json [hunk 1]
```diff
@@ -43,7 +44,9 @@
     "ethereumjs-util": "^6.0.0",
     "fast-memoize": "^2.5.1",
     "merkletreejs": "0.0.22",
-    "openzeppelin-solidity": "2.1.1"
+    "mongodb": "^3.1.13",
```
Comment (josojo) [inline package.json]: should this be devDependency?

Comment (bh2smith) [reply package.json]: Im not sure. It is necessary for the application to run.

Comment (bh2smith) [reply package.json]: No I think this is a standard dependency for the project to work. The dev-dependencies, are for things the travis build needs to run (i.e. like for the linter etc..)

**Review by josojo — APPROVED** [review]
Really good PR. Like all the additional tests

File: package.json [hunk 2]
```diff
@@ -45,14 +45,15 @@
     "fast-memoize": "^2.5.1",
     "merkletreejs": "0.0.22",
     "mongodb": "^3.1.13",
-    "openzeppelin-solidity": "2.1.1"
+    "openzeppelin-solidity": "2.1.1",
```
Comment (bh2smith) [inline package.json]:
```suggestion
    "openzeppelin-solidity": "2.1.1"
```

**Comment by fleupold** [comment]
This is awesome!

**Comment by bh2smith** [comment]
Right I forgot to mention this kickass thing! @fleupold

**Pull request merged**